text/0000-machine-readable-tests-output.md

```diff
@@ -0,0 +1,146 @@
+| `build` | MUST be ISO8601 timestamp of build.                            |
+| `name`  | OPTIONAL suite name i.e. "Tests", "Benchmarks" or "Doc-Tests". |
+| `count` | MUST be count of all tests (including ignored).                |
+| `rustc` | MUST be version of Rust compiler used to build test suite.     |
+
+### Test
+
+```json
+{
+  "type": "test",
+  "subtype": "should_panic",
+  "status": "ok",
+  "label": "octavo::digest::md5::tests::test_md5",
+  "file": "src/digest/md5.rs",
+  "line": 684,
+  "stdout": "",
+  "stderr": "",
+  "duration": 100
+}
+```
+
+Describes test. Each test MUST produce only one test object.
+
+Fields:
+
+| Name         | Description                                                                    |
+| ----         | -----------                                                                    |
+| `type`       | MUST be `test`.                                                                |
+| `subtype`    | SHOULD be one of `test`, `bench` or `should_panic`.                            |
+| `status`     | MUST be one of `ok`, `fail` or `ignore`.                                       |
+| `reason`     | MUST describe of failure if `status` is `fail`. Otherwise MUST NOT be present. |
+| `label`      | MUST be unique identifier of test.                                             |
+| `file`       | OPTIONAL file name containing test.                                            |
+| `line`       | OPTIONAL line in `file` that contains test.                                    |
+| `stdout`     | OPTIONAL output of standard output for given test.                             |
+| `stderr`     | OPTIONAL output of standard error for given test.                              |
+| `duration`   | OPTIONAL test execution duration in nanoseconds. MUST be present in benchmark. |
+| `throughput` | OPTIONAL test throughput in case of benchmark test.                            |
+
+### Final
+
+```json
+{
+  "type": "final",
+  "results": {
+    "ok": 10,
+    "fail": 0,
+    "ignore": 2
+  }
+}
+```
+
+Finish test suite and closes stream. Parser MUST ignore all other input unless
+it is new suite.
+
+Fields:
+
+| Name      | Description                                                                                                                                 |
+| ----      | -----------                                                                                                                                 |
+| `type`    | MUST be `final`.                                                                                                                            |
+| `results` | MUST be object containing only 3 fields: `ok`, `fail` and `ignore` which describe how many test passed, failed or was ignored respectfully. |
+
+
+# Drawbacks
+
+This is breaking change in tooling and will require new tool that will provide
+current functionality for compatibility reasons, but IMHO small pain for big gain.
```

Comment:
Why not make the machine-readable output an option and keep the human-readable output the default? Machine-readable output by default makes for a dreadful first-time experience.

FWIW pytest outputs a human-readable format to stdout and optionally a machine-readable one to a specified file.

The ability to output both human-readable and machine-readable is actually convenient in CI, you can send the human-readable output to a human-readable log and the machine-readable output to the CI's parser.